### PR TITLE
7345 export map public ui

### PR DESCRIPTION
**Pull request by gfiorav**
UI is done. Won't work due to back end API refusing connection without API key.

I still have to figure this one out.

cc/ @juanignaciosl @xavijam

### CODE REVIEW

**Comment by gfiorav** [comment]
Please CR @xavijam

**Comment by juanignaciosl** [comment]
You can test this merging #7404.

File: lib/assets/javascripts/cartodb/public_map/public_map_window.js [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
 
+  events: {
+    'click .js-Navmenu-link--download-map': '_exportMap'
+  },
+
+  _exportMap: function (e) {
+    e.preventDefault();
+
+    var view = new ExportMapView({
+      model: new cdb.admin.ExportMapModel({ 'visualization_id': vis_id }),
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/public_map/public_map_window.js]: `cdb.admin.ExportMapModel` or just `ExportMapModel`?

**Comment by xavijam** [comment]
I can't add anything, looks perfect, easy dizzy 👏

File: lib/assets/javascripts/cartodb/public_map/public_map_window.js (hunk 1 above)
Comment (gfiorav) [inline lib/assets/javascripts/cartodb/public_map/public_map_window.js]: Yep, changing!

**Comment by gfiorav** [comment]
Hey @xavijam , 2 things:

+ I couldn't change the constructor after all, errors ensued.
+ I had to check for `window.user_data !== undefined` for tracking since `window.user_data.email` isn't defined for public users.

**Comment by gfiorav** [comment]
Ok, @juanignaciosl, I've tested in staging and it looks 👌 

Can you proceed to merge the backend part?

**Comment by juanignaciosl** [comment]
I'd add a `export_map_public` or similar for Hubspot from the very beginning.

**Comment by xavijam** [comment]
That would be great :)

File: lib/assets/javascripts/cartodb/public_map/public_map_window.js [hunk 2]
```diff
@@ -1,14 +1,32 @@
 var $ = require('jquery-cdb-v3');
 var cdb = require('cartodb.js-v3');
 var PublicMap = require('./public_map_view');
-  
+var ExportMapView = require('../common/dialogs/export_map/export_map_view');
+var ExportMapModel = require('../models/export_map_model.js');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/public_map/public_map_window.js]: If you are not using ExportMapModel we could remove it right?

File: lib/assets/javascripts/cartodb/public_map/public_map_window.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/public_map/public_map_window.js]: Sorry commented above ^^^ :S

File: lib/assets/javascripts/cartodb/public_map/public_map_window.js (hunk 2 above)
Comment (gfiorav) [inline lib/assets/javascripts/cartodb/public_map/public_map_window.js]: I can't get rid of this without an error ensuing either 😞

**Comment by gfiorav** [comment]
I've added the hubspot metric trigger, to the app_config.yml and the cookbooks. 

I'll merge when 🍏 if no-one opposes.

**Comment by juanignaciosl** [comment]
💃 💃

**Pull request merged**